utils/to-valid-package-name.js

```diff
@@ -0,0 +1,21 @@
+const validate = require('validate-npm-package-name');
+
+module.exports = function toValidPackageName(packageName) {
+  const { validForNewPackages, validForOldPackages } = validate(packageName);
+  const isValid = validForNewPackages && validForOldPackages;
+
+  if (!isValid) {
+    // NOTICE: Create an always valid package name (disallow almost everything)
+    return packageName.toLowerCase()
+      // Remove all non "a-z", "0-9", "-" characters with hyphen.
+      .replace(/[^a-z0-9\\-]/g, '-')
+      // Remove hyphen sequence (> 1).
+      .replace(/-{2,}/g, '-')
+      // Remove leading and trailing hyphen.
+      .replace(/^-|-$/g, '')
+      // If empty, replace with my-lumber-project.
+      .replace(/^-*$/g, 'my-lumber-project');
+  }
```

Comment:
We are not sure the returned string is valid, right? 
If the warranty is the replacement, then we don't need a validate lib.

Why not 1/replace then 2/check validity and 3/ throw if it's not valid?

Reply:
Or maybe always return replacement and remove the lib?

Reply:
> maybe always return replacement and remove the lib?

I think we can keep the actual name without a drastic sanitization when it is valid: so it is returned as is if it's valid.

> We are not sure the returned string is valid, right?

Yes you are right. The new process is:

1. Validate it (if valid return it as is)
2. Sanitize it.
4. Validate it again (if valid return it, if not valid return `lumber-project`)